postman_problems/solver.py

```diff
@@ -1,105 +1,49 @@
     """
     Solving the RPP from beginning (load network data) to end (finding optimal route).  This optimization makes a
      relatively strong assumption: the starting graph must stay a connected graph when optional edges are removed.
     If this is not so, an assertion is raised.  This class of RPP generalizes to the CPP strategy.
-
     Args:
-        edgelist_filename (str): filename of edgelist.  See cpp.py for more details
-        start_node (str): name of starting node.  See cpp.py for more details
+        edgelist_filename (str): filename of edgelist.  
+        start_node (str): name of starting node.  
         edge_weight (str): name edge attribute that indicates distance to minimize in CPP
-        verbose (boolean): log info messages?
-
+        turn_weight_coefficient (float): turn weight coefficient used to add turn_weight attributes to g_full
     Returns:
-        tuple(list[tuple(str, str, dict)], networkx.MultiGraph]:
+        list[tuple(str, str, dict)]:
         Each tuple is a direction (from one node to another) from the CPP solution route.
           The first element is the starting ("from") node.
           The second element is the end ("to") node.
           The third element is the dict of edge attributes for that edge.
-        The original graph is returned as well.  This is needed for visualization
     """
 
-    logger_rpp.disabled = not verbose
+    el = read_edgelist(edgelist_filename)
 
-    logger_rpp.info('read edgelist')
-    el = read_edgelist(edgelist_filename, keep_optional=True)
-
-    logger_rpp.info('create full and required graph')
     g_full = create_networkx_graph_from_edgelist(el)
-    g_req = create_required_graph(g_full)
-    assert_graph_is_connected(g_req)
-
-    logger_rpp.info('getting odd node pairs')
-    odd_nodes = get_odd_nodes(g_req)
-    odd_node_pairs = list(itertools.combinations(odd_nodes, 2))
-
-    logger_rpp.info('get shortest paths between odd nodes')
-    odd_node_pairs_shortest_paths = get_shortest_paths_distances(g_full, odd_node_pairs, edge_weight)
-
-    logger_rpp.info('Find min weight matching using blossom algorithm')
-    g_odd_complete = create_complete_graph(odd_node_pairs_shortest_paths, flip_weights=True)
-    odd_matching = dedupe_matching(nx.algorithms.max_weight_matching(g_odd_complete, True))
-
-    logger_rpp.info('add the min weight matching edges to g')
-    g_aug = add_augmenting_path_to_graph(g_req, odd_matching)
-
-    logger_rpp.info('get eulerian circuit route')
-    circuit = list(create_eulerian_circuit(g_aug, g_full, start_node))
-
-    return circuit, g_full
-
+    
+    g_full, pos = sl.create_pos_and_add_to_graph(g_full, complete_g)
+    
+    g_full = create_turn_weight_edge_attr(g_full, length_weight='distance', normalization_coefficient=turn_weight_coefficient)
 
-def cpp(edgelist_filename, start_node=None, edge_weight='distance', verbose=False):
```

Comment:
Any to reason to remove the Chinese Postman Problem (CPP)?    If we were to bring the capability to support Directed Graphs in, we'd need to find a way to keep support for CPP

Reply:
No good reason. I was trying to streamline what code was necessary for my implementation, which I now am realizing was not the right approach.